akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1039,7 +1096,29 @@ abstract class ShardCoordinator(
           case None =>
             log.debug("Rebalance of non-existing shard [{}] is ignored", shard)
         }
+      }
+    }
+  }
 
+  def shutdownShards(shards: Set[ShardId]): Unit = {
```

Comment:
The only difference between continueRebalance and shutdownShards is again the logging, unless I missed something. I don't think we should duplicate the code just because of that. Better to pass `rebalance: Boolean` flag as a parameter.